File: services/python/starlette/src/application.py
import asyncio
from starlette.applications import Starlette
from starlette.middleware import Middleware
from starlette.routing import Route

from app.config import Config
from app.state import AppState
from routes.middleware import LogMiddleware, ChaosMiddleware, BodySizeMiddleware
from routes.handlers import (
    post_handler,
    get_handler,
    delete_handler,
    put_handler,
    patch_handler,
    internal_handler,
    invalid_method_handler,
    invalid_path_handler,
)

BYTES: int = 1024


class Application:
    def __init__(self) -> None:
        config = Config.new()
        self.app = Starlette(
            routes=[
                Route("/", endpoint=post_handler, methods=["POST"]),
                Route("/{id:uuid}", endpoint=get_handler, methods=["GET"]),
                Route("/{id:uuid}", endpoint=delete_handler, methods=["DELETE"]),
                Route("/{id:uuid}", endpoint=put_handler, methods=["PUT"]),
                Route("/{id:uuid}", endpoint=patch_handler, methods=["PATCH"]),
            ],
            middleware=[
                Middleware(LogMiddleware),
                Middleware(ChaosMiddleware),
                Middleware(
                    BodySizeMiddleware,
                    max_size=config.request_size_limit * BYTES,
                ),
            ],
            exception_handlers={
                404: invalid_path_handler,
                405: invalid_method_handler,
                Exception: internal_handler,
            },
        )

        self.app.state.config = config
        self.app.state.app_state = AppState()
        self.app.state.tasks_lock = asyncio.Lock()

File: services/go/fiber/application.go
package main

import (
	"app"
	"fiber/routes/handlers"
	"fiber/routes/middleware"

	"github.com/gofiber/fiber/v3"
	recoverer "github.com/gofiber/fiber/v3/middleware/recover"
)

const BYTES = 1024

type Application struct {
	Engine *fiber.App
	state  *app.AppState
}

func NewApplication(appConfig app.AppConfig) *Application {
	state := app.NewState(appConfig)

	engine := fiber.New(fiber.Config{
		BodyLimit:    int(appConfig.RequestSizeLimit * BYTES),
		ErrorHandler: handlers.ErrorHandler,
	})

	engine.Use(recoverer.New())
	engine.Use(middleware.LogMiddleware())
	engine.Use(middleware.ChaosMiddleware(state))

	engine.Post("/", handlers.PostHandler(state))
	engine.Get("/:id", handlers.GetHandler(state))
	engine.Delete("/:id", handlers.DeleteHandler(state))
	engine.Patch("/:id", handlers.PatchHandler(state))
	engine.Put("/:id", handlers.PutHandler(state))

	return &Application{Engine: engine, state: state}
}
